### ISSUE TITLE
cb_explore_adf does not work with simple CB data format on macOS

### ISSUE THREAD

**Issue by sumeet-iitg** [VowpalWabbit/vowpal_wabbit#2103]
#### Describe the bug
cb_explore_adf does not work with simple CB data format on macOS

#### To Reproduce
1. vw --cb_explore_adf --epsilon 0.05 -d ./examples/text_highnoise_m.vw.txt -f cb_adf_model.vw
2. Data (Text_highnoise also used in Tutorial)is in basic CB input format: act_id | w0:val0 w1:val2 ...
  Downloaded from [here] (https://raw.githubusercontent.com/zcc1307/vw_datasets/master/datasets/text_highnoise_m.vw)

#### Expected behavior
Should show loss etc.

#### Observed Behavior
final_regressor = cb_adf_model.vw
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = ./examples/text_highnoise_m.vw.txt
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
    n.a.     n.a.            1            1.0  unknown        0:0.0001...  1010000

finished run
number of examples = 1
weighted example sum = 1.000000
weighted label sum = 0.000000
average loss = n.a.
total feature number = 1010000

#### Environment
What version of VW did you use?
VW --version 
8.7.0 (git commit: 00660f7a)
Installed from github source and running brew install commands.
Using macOS 10.14.

**Comment by jackgerrits** [comment]
When using `--cb_explore_adf` you must use the adf label format, since there is an unknown number of actions. 

In the CB tutorial, under section 3 of [input format](https://vowpalwabbit.org/guides/contextual_bandits.html#input-format) it describes the required format. You'll notice that the format differs from section 1 which is the format you are supplying.

**Comment by sumeet-iitg** [comment]
Okay, I had seen that. But somehow thought this was not required. Probably got confused by the Warm cb tutorial dataset which uses --cb_explore_adf: https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Warm-starting-contextual-bandits

Follow up question, As per --cb_explore_adf format we need to specify one action per line and the action number is decided by the line number, then does it mean we have to provide a line for every action, until the newline?

**Comment by jackgerrits** [comment]
Yes, the newline is used to separate a grouping of shared+actions. So it is one line (optionally) for shared, then a line for every action, then a newline.

```
shared | s_1 s_2                  <--- Shared
0:1.0:0.5 | a:1 b:1 c:1           <--- Action 0
| a:0.5 b:2 c:1                      <--- Action 1
                                           <--- New line
shared | s_1 s_2                  <--- Shared
0:1.0:0.5 | a:1 b:1 c:1           <--- Action 0
| a:0.5 b:2 c:1                      <--- Action 1
```

**Comment by sumeet-iitg** [comment]
Thank you so much for the clarification. Any comment on the input format mentioned in the warm cb tutorial which used --cb_explore_adf?

**Comment by jackgerrits** [comment]
I'll have a look at it and get back to you

**Comment by jackgerrits** [comment]
It seems the misunderstanding may be because warm start uses multiclass labels, even though under the covers it is using the cb_explore_adf algorithm for that wiki article.

Since VW uses a stack of reductions, data passes through warm cb before it gets to cb_explore_adf. Warm cb handles converting it to the format cb_explore_adf understands.

**Comment by sumeet-iitg** [comment]
Okay, that makes sense. Thank you so much for the explanation. 
Closing the issue.
